registry/config.go

```diff
@@ -143,6 +143,13 @@ func (config *serviceConfig) LoadInsecureRegistries(registries []string) error {
 
 skip:
 	for _, r := range registries {
+		if strings.Contains(r, "://") {
+			// Insecure registry does not contain a scheme.
+			// before returning err, roll back to original data
+			config.ServiceConfig.InsecureRegistryCIDRs = originalCIDRs
+			config.ServiceConfig.IndexConfigs = originalIndexInfos
+			return fmt.Errorf("insecure registry %s should not contain a scheme '%s://'", r, strings.Split(r, "://")[0])
```

Comment:
Looking at this again; overall, I'd prefer explicit over implicit; however, since we did accept insecure registries _with_ scheme in the past, so I don't think we can error out here, as it will break existing setups that upgrade to a new version of docker.

This should probably be a _warning_, and we should strip the scheme in the configuration.